src/model.ts

```diff
@@ -86,6 +86,12 @@ const assignProps = (target: any, source: any) => {
   }
 };
 
+const defaultThrowAction = (name: string) => {
+  return () => {
+    throw new Error(`Can't run action "${name}" for a readonly instance`);
```

Comment:
Can I be annoying and suggest we use an Error subclass for maximum descriptiveness and catch-ability